generated/language/src/test/java/com/google/cloud/language/v1/spring/LanguageAutoConfigurationTests.java

```diff
@@ -0,0 +1,350 @@
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.boot.autoconfigure.AutoConfigurations;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.threeten.bp.Duration;
+
+@ExtendWith(MockitoExtension.class)
+class LanguageAutoConfigurationTests {
+
+  private static final String SERVICE_CREDENTIAL_LOCATION =
+      "src/test/resources/fake-credential-key.json";
+  private static final String TOP_LEVEL_CREDENTIAL_LOCATION =
+      "src/test/resources/fake-credential-key-2.json";
+  private static final String SERVICE_CREDENTIAL_CLIENT_ID = "45678";
+  private static final String TOP_LEVEL_CREDENTIAL_CLIENT_ID = "12345";
+  private static final String SERVICE_OVERRIDE_CLIENT_ID = "56789";
+
+  @Mock private TransportChannel mockTransportChannel;
+  @Mock private ApiCallContext mockApiCallContext;
+  @Mock private TransportChannelProvider mockTransportChannelProvider;
+  @Mock private CredentialsProvider mockCredentialsProvider;
+
+  private ApplicationContextRunner contextRunner =
+      new ApplicationContextRunner()
+          .withConfiguration(
+              AutoConfigurations.of(
+                  GcpContextAutoConfiguration.class, LanguageServiceSpringAutoConfiguration.class));
+
+  @Test
+  void testLanguageServiceClientCreated() {
+    this.contextRunner.run(
+        ctx -> {
+          LanguageServiceClient client = ctx.getBean(LanguageServiceClient.class);
+          assertThat(client).isNotNull();
+        });
+  }
+
+  @Test
+  void testCredentials_fromServicePropertiesIfSpecified() {
+    this.contextRunner
+        .withPropertyValues(
+            "spring.cloud.gcp.credentials.location=file:" + TOP_LEVEL_CREDENTIAL_LOCATION,
+            "com.google.cloud.language.v1.language-service.credentials.location=file:"
+                + SERVICE_CREDENTIAL_LOCATION)
+        .run(
+            ctx -> {
+              LanguageServiceClient client = ctx.getBean(LanguageServiceClient.class);
+              Credentials credentials =
+                  client.getSettings().getCredentialsProvider().getCredentials();
+              assertThat(((ServiceAccountCredentials) credentials).getClientId())
+                  .isEqualTo(SERVICE_CREDENTIAL_CLIENT_ID);
+            });
+  }
+
+  @Test
+  void testCredentials_fromTopLevelIfNoServiceProperties() {
+    this.contextRunner
+        .withPropertyValues(
+            "spring.cloud.gcp.credentials.location=file:" + TOP_LEVEL_CREDENTIAL_LOCATION)
+        .run(
+            ctx -> {
+              LanguageServiceClient client = ctx.getBean(LanguageServiceClient.class);
+              Credentials credentials =
+                  client.getSettings().getCredentialsProvider().getCredentials();
+              assertThat(((ServiceAccountCredentials) credentials).getClientId())
+                  .isEqualTo(TOP_LEVEL_CREDENTIAL_CLIENT_ID);
+            });
+  }
+
+  @Test
+  void testShouldUseDefaultTransportChannelProvider() {
```

Comment:
For the sake of coverage, you might also want to add a `testCustomTransportChannelProviderSetToRest`? 
when property `"com.google.cloud.language.v1.language-service.useRest=true"` present,
we should expect the `transportChannelProvider` set to 
```
LanguageServiceSettings.defaultHttpJsonTransportProviderBuilder().build();
```

Reply:
Thank you for this suggestion and added! I think it also addresses what I missed earlier in https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/1355#discussion_r1052341540.